feat(sdp):  Adding the hive shell to standalone dns proxy

vipul-21 opened the pull request:
Adding the hive shell to standalone dns proxy. It will allow the health cmd be used for liveliness and readiness probes on the daemonset.

> azureuser@vipul-vm:~/ws/cilium$ k exec -it -n kube-system          standalone-dns-proxy-hz7vq      -- standalone-dns-proxy shell health
health
└── job-module-status-metrics                   [OK] Running (46m, x1)

<img width="1285" height="456" alt="image" src="https://github.com/user-attachments/assets/19564fac-4e0b-47db-8d5f-fbfffed9936a" />


```release-note
feat(sdp):  Adding the hive shell to standalone dns proxy
```

vipul-21 commented:
/test

squeed commented:
Allocating a local port can be risky, since this is in the host's network namespace. Can this be a local domain socket instead? Then you can use an `exec` probe.

Even more awesome would be to use the [Hive Shell](https://github.com/cilium/hive/tree/main/shell), which exposes a standardized debug interface. Then you add a shell client to the sdp binary.

vipul-21 committed:
feat(sdp): Adding the hive shell to standalone dns proxy

Provides the shell cmd for sdp. It will allow the health
cmd be used for liveliness and readiness probes on the daemonset.

Signed-off-by: Vipul Singh <singhvipul@microsoft.com>

vipul-21 commented:
@squeed Thank you for the recommendation to use hive shell. I removed the server/api code and added the shell to sdp. Wondering if the readiness/liveliness probes will be:
```
livenessProbe:
  exec:
    command:
    - standalone-dns-proxy
    - shell
    - health
```
A bit confused about the default health shell provides. This the hive health right ?
```
azureuser@vipul-vm:~/ws/cilium$ k exec -it -n kube-system          standalone-dns-proxy-hz7vq      -- standalone-dns-proxy shell health
health
└── job-module-status-metrics                   [OK] Running (46m, x1)
```

If so, can you also remove the api/build reviewers as this might be not relevant now. Thanks!

bimmlerd reviewed (commented):
LGTM as is, do you plan on adding more commands to run in the shell in separate PRs?

squeed commented:
/test

vipul-21 commented:
/ci-gateway-api

vipul-21 commented:
/test

vipul-21 commented:
/ci-l7
